File: game_client/Assets/Scripts/ResScripts/NPResCommon/HeroCondition/ConditionObj/HeroCondition_CS_ATTR.cs
using ALPackage;
using Common.ConditionEnum;
using CommonEnum;

namespace GOE
{
    /// <summary>
    /// 判断大臣是否拥有该特长 CS_ATTR:EBasicAttrType
    /// </summary>
    public class HeroCondition_CS_ATTR : _ABasicHeroCondition
    {
        private ESpecAttrType _m_eBasicAttrType;//特长枚举
        
        public override EHeroConditionType conditionType { get { return EHeroConditionType.CS_ATTR; } }
        
        /// <summary>
        /// 读取条件信息
        /// </summary>
        /// <param name="_reader"></param>
        /// <returns></returns>
        public static HeroCondition_CS_ATTR readStr(ALStringReader _reader)
        {
            if (_reader == null)
            {
                UnityEngine.Debug.LogError("Can not read str for EHeroConditionType.CS_ATTR _reader == null");
                return null;
            }
            
            string attrTypeStr = _reader.readItem(':');
            if (string.IsNullOrEmpty(attrTypeStr))
            {
                UnityEngine.Debug.LogError("Can not read str for EHeroConditionType.CS_ATTR[" + _reader.srcString + "]");
                return null;
            }

            HeroCondition_CS_ATTR cond = new HeroCondition_CS_ATTR();

            ALCommon.TryEnumParse(typeof(ESpecAttrType), attrTypeStr, out cond._m_eBasicAttrType);
            return cond;
        }
        
        public override bool isEnable(HeroRefObj _data, HeroConditionVarInfo _varVariableInfo)
        {
            if (_data == null)
                return false;

#if NP_GAME
            return _data.spec_attr_type == _m_eBasicAttrType;
#endif
            return false;
        }
    }
}

File: game_server/GameRes/src/NPGameRes/GameObjs/HeroCondition/ConditionObj/HeroCondition_CS_ATTR.java
package NPGameRes.GameObjs.HeroCondition.ConditionObj;

import Common.ConditionEnum.EHeroConditionType;
import CommonEnum.ESpecAttrType;
import NPCommon.CommonObj.NPStringReader;
import NPCommon.Log.CommLog;
import NPGameRes.GameObjs.HeroCondition._ABasicHeroCondition;

public class HeroCondition_CS_ATTR extends _ABasicHeroCondition
{
    private ESpecAttrType _m_attr = ESpecAttrType.NONE;

    public ESpecAttrType attrType()
    {
        return _m_attr;
    }

    @Override
    public EHeroConditionType conditionType()
    {
        return EHeroConditionType.CS_ATTR;
    }

    public static HeroCondition_CS_ATTR readCond(NPStringReader _reader)
    {
        HeroCondition_CS_ATTR cond = new HeroCondition_CS_ATTR();

        String attr = _reader.readItem();
        if (null == attr)
        {
            CommLog.error("Can not read str for EHeroConditionType.CS_ATTR[" + _reader.getSrcString() + "]");
            return null;
        }

        cond._m_attr = ESpecAttrType.valueOf(attr.toUpperCase());

        return cond;
    }
}
